### ISSUE TITLE
Device specific `GroupByRegister` throws an error on private access addresses

### ISSUE THREAD

**Issue by bruno-f-cruz** [bonsai-rx/harp#72]
When saving registers from a known Harp Device, the following pattern should be used:

![image](https://github.com/bonsai-rx/harp/assets/7049351/d49d7fe1-56f7-4504-bada-65f01f60a698)

Since the `GroupbyRegister` namespace is specific to the device in question, it assumes a known dictionary of `Address(int)` to register name. This assignment fails when:

 - A harp message from a private address is issued from the device (e.g.: a `reserved` address). While this seems unlikely, the most common case is the op-control issued read dump which returns a volley of all defined addresses in the microcontroller, including the private ones.
 - When a `GroupByRegister` is used for the wrong device that has an address number that does not exist in the current device's schema.

I would argue that either case, the device-specific `GroupBy` operator should drop all messages that do not have a corresponding key. The rationale is that we are already pushing the responsibility of using the correct device namespace to the user (Via the `Device` node that checks the `WhoAmI` and the corresponding `GroupByRegister` added right after). Similarly, if a register is defined as `private` in the schema, and thus not part of the list of registers available in the interface, it should, by definition, not be exposed to the user and thus not propagated to the device-specific logger. If the user **really** wants to use those private addresses, we should assume that a bespoke functionality is being attempted and the device-specific `GroupByRegister` should be replaced by the its Bonsai.Harp generic counterpart, that does not rely on a key assignment.

**Comment by bruno-f-cruz** [comment]
A temporary solution to circumvent this problem is to exclude all messages in the sequence that are not represent in the Register maps. E.g.:

```CS
using Bonsai;
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Linq;
using System.Reactive.Linq;
using Bonsai.Harp;

namespace Harp.Behavior
{
    [Combinator]
    [Description("Filters out messages that are not in the Device register map")]
    [WorkflowElementCategory(ElementCategory.Combinator)]
    public class FilterVisibility
    {
        IReadOnlyDictionary<int, Type> CoreRegisterMap = Bonsai.Harp.Device.RegisterMap;

        IReadOnlyDictionary<int, Type> RegisterMap = Device.RegisterMap;

        public IObservable<HarpMessage> Process(IObservable<HarpMessage> source)
        {
            return source.Where(message => {
                return RegisterMap.ContainsKey(message.Address) | CoreRegisterMap.ContainsKey(message.Address);
            });
        }
    }
}
```

**Comment by glopesdev** [comment]
@bruno-f-cruz @filcarv I'm not sure if this was mentioned before, but somehow I feel that the correct behavior for private or reserved registers would be for the microcontroller to not send them at all through the USB interface. If they are private / non-visible they should never be exposed to the high-level, including in the initial state core dump.

Not sure how hard this would be to implement in the firmware, but given this usually happens only once at the beginning hopefully it wouldn't have much of a performance impact.
